diff --git a/node_modules/@lunarr/vlc-player/android/build.gradle b/node_modules/@lunarr/vlc-player/android/build.gradle
index e2c7c49..d38d3fe 100644
--- a/node_modules/@lunarr/vlc-player/android/build.gradle
+++ b/node_modules/@lunarr/vlc-player/android/build.gradle
@@ -21,7 +21,7 @@ android {
     buildTypes {
         release {
             minifyEnabled false
-            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
     }
 }
